src/db/user.ts

```diff
@@ -0,0 +1,39 @@
+import { ObjectId } from 'mongodb';
+import { UserIdentity, userIdentitySchema } from './user-identity';
+import { Schema, Document } from 'mongoose';
+/**
+ * This is an entry for a user, this is normalized
+ * for "future" proofing future features that will
+ * deal with the discord-user.
+ */
+export interface User extends Document {
+  /**
+   * The discord server userId
+   */
+  userId: string | ObjectId;
+  /**
+   * List of previous "identity" information
+   * for a given user. Used to keep track of
+   * users incase they change their discord user name.
+   */
+  identities: Array<UserIdentity>;
+}
```

Comment:
So this will be a separate model from the user's suspension history?

Reply:
So the idea I had was the `User` would be used to identify user's from Discord and possibly FCC itself. 

The suspension history/logging history/whatever information will be saved in other schemas by itself. This should make the database more flexible if we end up adding more stuff. Everything will have to have the same reference to the original user.

**I wasn't sure about the `userId` being different than the mongodb generated _id**, and was thinking if we should maybe make them 1 in the same?

Reply:
Discord ID values are theoretically unique, so that shouldn't be a problem. The only concern I would have there is future-proofing (if Discord runs out of numbers or switches how they identify a user, we'd have to change the `_id` value and I'm not sure mongo likes that).